firrtl/src/test/scala/firrtlTests/SerializerSpec.scala

```diff
@@ -82,6 +82,50 @@ object SerializerSpec {
       "test"
     )
 
+  val probeModule: String =
+    """module probeModule :
+      |  input clock : Clock
+      |  input cond : UInt<1>
+      |  input in : UInt<8>
+      |  input inProbe : Probe<UInt<8>>
+      |  output out : UInt<8>
+
+      |  inst c of child
+      |  wire fooProbe : Probe<UInt<8>>
+      |  c.in <= in
+      |  define fooProbe = probe(c.in)
+      |  out <= read(out)
+      |  force_initial(inProbe, UInt<8>("h64"))
```

Comment:
1) Input references aren't really a thing that can be used, but they're not rejected entirely if they're used safely.  Unfortunately can't force up an input reference unless it resolves to below you, which is impossible here.

2) Force (and release) only take RWProbe's.  Again unsure whether this is something to check in Chisel, so don't mean to nit.

It's a little messy and geared for compiler bugs FWIW but various examples/sample inputs of ref types in FIRRTL can be found [here](https://github.com/dtzSiFive/reftype-inputs), with examples from the specification proposal in the 'spec' subdirectory.  Some are not complete/valid as standalone designs, but might be useful as a.. reference.  Generated as finding edge cases and breaking the existing and under-development implementation of these things.